### PR TITLE
Added default agent tags

### LINKED ISSUE #632: Inject environment information into JAEGER_TAGS env variable
The operator could get environment information (e.g. namespace) and inject it into environmental variable `JAEGER_TAGS` which can be used by jaeger agent to add tags to each process/span reporting to the collector.

Related issue https://github.com/jaegertracing/jaeger/issues/1755

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #632.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
@objectiser are the go.mod/go.sum changes in line with the ones you observed when running `./.ci/script.sh` locally ?

**Comment by objectiser** [comment]
@jpkrohling yes looks the same from a quick scan.

**Comment by jpkrohling** [comment]
> yes looks the same from a quick scan

Need to save some time to properly learn go mod. I thought the dependency list was static, unless we actively ran an update command.

In any case, I'll keep the dependency changes in this PR, as it will otherwise fail due to the unstaged changes to the go.mod/sum files, even though no dependencies actually changed as part of this PR.

File: pkg/inject/sidecar_test.go [hunk 1]
```diff
@@ -252,12 +282,37 @@ func TestSidecarOrderOfArguments(t *testing.T) {
...
+
+	// test
+	dep = Sidecar(jaeger, dep)
+
+	// verify
+	assert.Len(t, dep.Spec.Template.Spec.Containers, 2)
+	assert.Contains(t, dep.Spec.Template.Spec.Containers[1].Args, "--jaeger.tags=key=val")
+}
+
+func containsOptionWitPrefix(t *testing.T, args []string, prefix string) bool {
```
Comment (objectiser) [inline pkg/inject/sidecar_test.go]: minor: `containsOptionWithPrefix`

Comment (jpkrohling) [reply pkg/inject/sidecar_test.go]: Good catch!

**Pull request merged**